uor-framework/uor-client-go
# Feat/add sigstore

Sigstore integration! 

Signing occurs during the push phase of Collection publishing. Set the `--sign` flag to enable signing during publishing. 

Signature verification occurs by default during Collection retrieval. Set the `--no-verify=true` flag to disable signature verification when pulling content from a UOR Collection.  

Test Instructions:

Step 1: Build a schema or any collection. 
Step 2: Push the collection to a remote with signing disabled. Example: `uor-client-go push my-workspace localhost:5000/test/collection:latest`
Step 3. Pull the schema and the pull should fail. Example:  `uor-client-go pull  localhost:5000/test/collection:latest -o test/`
Step 4: Push and sign the collection.  Example: `uor-client-go push my-workspace localhost:5000/test/collection:latest --sign`
Step 5: Pull the collection and verify signatures: `uor-client-go pull  localhost:5000/test/collection:latest -o test/`
Step 6: Pull the collection and do not verify signatures: `uor-client-go pull  localhost:5000/test/collection:latest -o test/ --no-verify=true`
